rdr_service/genomic/genomic_job_controller.py

```diff
@@ -1905,6 +1905,28 @@ def check_results_withdrawals(self):
 
         self.job_result = GenomicSubProcessResult.SUCCESS
 
+    def check_aw3_ready_missing_files(self):
+        """ Runs report to email list of samples that are ready for AW3 manifest generation but missing data files"""
+        genomics_dao = GenomicQueriesDao()
+        notification_email_address = config.getSettingJson(config.RDR_GENOMICS_NOTIFICATION_EMAIL, default=None)
+
+        array_missing_data = genomics_dao.get_aw3_array_records(return_missing_files=True)
+        wgs_missing_data = genomics_dao.get_aw3_wgs_records(return_missing_files=True)
+        if notification_email_address and any((array_missing_data, wgs_missing_data)):
+            message = 'The following samples matched AW3 manifest criteria except for the data file count:\n\n'
+            message += 'sample_id,genome_type\n'
+            message += '\n'.join([f'{f[1]},aou_array' for f in array_missing_data])
+            message += '\n'.join([f'{f[1]},aou_wgs' for f in wgs_missing_data])
+            EmailService.send_email(
+                Email(
+                    recipients=[notification_email_address],
+                    subject='AW3 ready samples with missing data files',
+                    plain_text_content=message
+                )
+            )
+        self.job_result = GenomicSubProcessResult.SUCCESS
+
+
```

Comment:
Sorry, why are we doing this? I thought the point of this feature set was to backfill the data files on the metrics records, not sure why we are concerned with missing files anymore.

Reply:
The idea behind this report is to catch any samples that don't have all of the data files so we don't build up a backlog if a file doesn't get uploaded or fails to be added to the database.

Reply:
The idea from what I understand is to alleviate bottlenecks in this process of the pipeline. One of those bottlenecks is checking for the data files in the buckets. The idea should just be to build the paths to the files and store them on the metrics records and not keep track of anything missing cause we are trying to take that responsibility off of the pipeline.

Reply:
For reference, during AW3 generation, we still need to check that we don't send Broad records with missing files. This was Megan's requirement, so we're still using gc_data_file to check that there are at least the number of required files there. The weekly report is just to say "Here's records that should be in an AW3 but don't have enough data files." It won't be updating data or setting any flags.